omod/src/test/java/org/openmrs/module/fhir2/providers/ObservationFhirResourceProviderWebTest.java

```diff
@@ -138,18 +140,29 @@ public void shouldGetObservationsByPatientUuid() throws Exception {
 		
 		verify(observationService).searchForObservations(isNull(), patientCaptor.capture(), isNull(), isNull());
 		assertThat(patientCaptor.getValue(), notNullValue());
-		assertThat(patientCaptor.getValue().getIdPart(), equalTo(PATIENT_UUID));
-		assertThat(patientCaptor.getValue().getResourceType(), equalTo("Patient"));
+		assertThat(patientCaptor.getAllValues().iterator().next().getValuesAsQueryTokens().iterator().next()
+		        .getValuesAsQueryTokens().iterator().next().getIdPart(),
+		    equalTo(PATIENT_UUID));
+		assertThat(patientCaptor.getAllValues().iterator().next().getValuesAsQueryTokens().iterator().next()
+		        .getValuesAsQueryTokens().iterator().next().getResourceType(),
+		    equalTo("Patient"));
 	}
 	
 	@Test
 	public void shouldGetObservationsByPatientIdentifier() throws Exception {
-		verifyUri("/Observation?subject.identifier=M4001-1");
+		verifyUri("/Observation?subject.identifier=M4001-1,ABS098,YT56RE,IU23O");
```

Comment:
So for these tests, rather than repurposing them for `OR` queries, we should leave the initial test as is, then test for `OR` and then add a separate test for `AND` (which looks like `/Observation?subject.identifier=M4001-1&subject.identifier=IU230`).

Also, it will be more readable if you save the `ValueAsQueryTokens` intermediate steps. Also note that unlike the results we sometimes use which are `Collections` the result of `getValuesAsQueryTokens()` are lists that can be indexed numerically.

Reply:
> (which looks like /Observation?subject.identifier=M4001-1&subject.identifier=IU230).

@ibacher Can you explain what is this expected to give. I am asking this because when I was testing for `/Observation?subject.identifier=M4001-1,IU230` I get size 2 of the list and when testing for `/Observation?subject.identifier=M4001-1&subject.identifier=IU230` I get size equal to 1

Reply:
So it depends on where you're looking. Remember that the hierarchy for these types goes:

Each `AndParam` contains 1 or more `OrParam`s which contains 1 or more `Param`s.

This means that if you search for `/Observation?subject.identifier=M4001-1&subject.identifier=IU23`, then the length of `patientCaptor.getValue().getValueAsQueryTokens()` should be 2, but the length of `patientCaptor.getValue().getValueAsQueryTokens().get(0).getValueAsQueryTokens` should be 1.